### PR TITLE
Allow adding resources from a system

### PR DESCRIPTION
**Pull request by torkleyy**
This change is [Reviewable](https://reviewable.io/reviews/slide-rs/specs/433)

### CODE REVIEW

**Comment by torkleyy** [comment]
bors delegate=Rhuagh

**Comment by Rhuagh** [comment]
I'll let others comment before pushing it.

File: src/world/lazy.rs [hunk 1]
```diff
@@ -252,21 +254,69 @@ impl LazyUpdate {
...
+    ///         for entity in ent.join() {
+    ///             lazy.exec_mut(move |world| {
+    ///                 // complete extermination!
+    ///                 world.delete_all();
+    ///             });
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    pub fn exec_mut<F>(&self, f: F)
```
Comment (WaDelma) [inline src/world/lazy.rs]: Do we really need separate mutable execute method? Can't we just promote normal execute to this? Just wondering.

File: src/world/mod.rs [hunk 2]
```diff
@@ -546,7 +546,10 @@ impl World {
             self.delete_components(&deleted);
         }
 
-        self.write_resource::<LazyUpdate>().maintain(&*self);
+        // we need to swap the queue out to be able to reborrow self mutable here
+        let mut lazy = self.write_resource::<LazyUpdate>().take();
+        lazy.maintain(&mut *self);
+        self.write_resource::<LazyUpdate>().restore(lazy);
```
Comment (WaDelma) [inline src/world/mod.rs]: This case of ownership dance is bit unfortunate, but it's most likely required.

File: src/world/lazy.rs (hunk 1 above)
Comment (torkleyy) [reply src/world/lazy.rs]: I think I used this to make this non-breaking.

File: src/world/mod.rs (hunk 2 above)
Comment (torkleyy) [reply src/world/mod.rs]: Yeah :(

**Comment by torkleyy** [comment]
bors r=Xaeroxe

**Comment by Rhuagh** [comment]
Ah, sorry, forgot about this one :( Too many things on my TODO.

**Comment by torkleyy** [comment]
Oh, np, it needed rebasing anyways.

**Pull request merged**